kv/kvserver: TestStoreRangeMergeConcurrentRequests failed

kv/kvserver.TestStoreRangeMergeConcurrentRequests [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286535&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286535&tab=artifacts#/) on master @ [57173569d3892064bde592fee1fb119d64bf7d8b](https://github.com/cockroachdb/cockroach/commits/57173569d3892064bde592fee1fb119d64bf7d8b):


```
=== RUN   TestStoreRangeMergeConcurrentRequests
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeConcurrentRequests734036705
    test_log_scope.go:74: use -show-logs to present logs inline
    client_merge_test.go:2374: /System/" aaa-testing": merge unexpected error: kv/kvserver/replica_command.go:791: merge failed: waiting for all left-hand replicas to initialize: operation "wait for replicas init" timed out after 5s: could not dial n1: context deadline exceeded
E210811 07:14:41.784119 157273 kv/kvserver/consistency_queue.go:188  [n1,consistencyChecker,s1,r12/1:/Table/1{6-7}] 1  computing own checksum: could not dial node ID 1: failed to connect to n1 at 127.0.0.1:34963: stopped
E210811 07:14:41.784221 157273 kv/kvserver/queue.go:1098  [n1,consistencyChecker,s1,r12/1:/Table/1{6-7}] 2  computing own checksum: could not dial node ID 1: failed to connect to n1 at 127.0.0.1:34963: stopped
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeConcurrentRequests734036705
--- FAIL: TestStoreRangeMergeConcurrentRequests (5.28s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestStoreRangeMergeConcurrentRequests PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/kv nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeConcurrentRequests.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -2262,6 +2262,7 @@ func TestStoreRangeMergeCheckConsistencyAfterSubsumption(t *testing.T) {
 func TestStoreRangeMergeConcurrentRequests(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderStress(t, "falls apart with timeouts under high concurrency")
 
 	var store *kvserver.Store
 	manualClock := hlc.NewHybridManualClock()

